### ISSUE TITLE
Testnet stuck @744147

### ISSUE THREAD

**Issue by roman-khimov** [nspcc-dev/neo-go#2284]
```
2021-12-01T09:10:13.053+0300    WARN    blockQueue: failed adding block into the blockchain     {"error": "onPersist failed: VM has failed: at instruction 0 (SYSCALL): insufficient funds", "blockHeight": 744146, "nextIndex": 744147}
```

**Comment by roman-khimov** [comment]
There is only 0x6f68451e814804c5e8ae705e35cb0639ecbb59a2dcf167aea2b8441709a76642 transaction in this block with an oracle response, but oracle contract account (NTz4UrybSL4C7HSfaVpXV6hcsWTkks8Nrj) doesn't have enough GAS to pay for it. That's because of the difference in handling of oracle request transaction.

C#:
```
{
   "result" : {
      "txid" : "0xdb6d66e66c07964a5c2fe339ad1a83cb0fdf63045aa246c6ff43995464a2638c",
      "executions" : [
         {
            "notifications" : [
               {
                  "state" : {
                     "type" : "Array",
                     "value" : [
                        {
                           "value" : "ogVaOQnUi7QT2zJ14HwkUtzY6/k=",
                           "type" : "ByteString"
                        },
                        {
                           "type" : "ByteString",
                           "value" : "m3m7LI7CP4hiMFkTzn4jK9PQ+BI="
                        },
                        {
                           "type" : "Integer",
                           "value" : "5000000"
                        }
                     ]
                  },
                  "contract" : "0xd2a4cff31913016155e38e474a2c06d08be276cf",
                  "eventname" : "Transfer"
               },
               {
                  "contract" : "0xd2a4cff31913016155e38e474a2c06d08be276cf",
                  "state" : {
                     "type" : "Array",
                     "value" : [
                        {
                           "type" : "Any"
                        },
                        {
                           "type" : "ByteString",
                           "value" : "WIcXEX4KqBByr6tx0t2J/nxLkv4="
                        },
                        {
                           "value" : "39000000",
                           "type" : "Integer"
                        }
                     ]
                  },
                  "eventname" : "Transfer"
               },
               {
                  "eventname" : "OracleRequest",
                  "contract" : "0xfe924b7cfe89ddd271abaf7210a80a7e11178758",
                  "state" : {
                     "value" : [
                        {
                           "type" : "Integer",
                           "value" : "17024"
                        },
                        {
                           "value" : "m3m7LI7CP4hiMFkTzn4jK9PQ+BI=",
                           "type" : "ByteString"
                        },
                        {
                           "value" : "aHR0cHM6Ly9hcGkubmVvZGVwb3Qub3JnL2FwaS84aEhWcGF1UVFZeGZTQ1FvZE5hNnBOTTNKZXVyQXFHUnBNeXhzOFI5aXZ0VlVjVnQ2YzlDeUJIS0dMeA==",
                           "type" : "ByteString"
                        },
                        {
                           "value" : "JC5yZXN1bHRbMF0uZGF0YS5jaGlsZHJlblswXS5kYXRh",
                           "type" : "ByteString"
                        }
                     ],
                     "type" : "Array"
                  }
               }
            ],
            "exception" : null,
            "gasconsumed" : "50784867",
            "stack" : [
               {
                  "value" : true,
                  "type" : "Boolean"
               }
            ],
            "trigger" : "Application",
            "vmstate" : "HALT"
         }
      ]
   },
   "jsonrpc" : "2.0",
   "id" : 1
}
```
Go:
```
{
   "id" : 1,
   "jsonrpc" : "2.0",
   "result" : {
      "executions" : [
         {
            "stack" : [],
            "exception" : "at instruction 29 (SYSCALL): failed native call: at instruction 6018 (ABORT): ABORT",
            "trigger" : "Application",
            "notifications" : [
               {
                  "eventname" : "Transfer",
                  "contract" : "0xd2a4cff31913016155e38e474a2c06d08be276cf",
                  "state" : {
                     "value" : [
                        {
                           "type" : "ByteString",
                           "value" : "ogVaOQnUi7QT2zJ14HwkUtzY6/k="
                        },
                        {
                           "type" : "ByteString",
                           "value" : "m3m7LI7CP4hiMFkTzn4jK9PQ+BI="
                        },
                        {
                           "value" : "5000000",
                           "type" : "Integer"
                        }
                     ],
                     "type" : "Array"
                  }
               }
            ],
            "gasconsumed" : "1078598",
            "vmstate" : "FAULT"
         }
      ],
      "txid" : "0xdb6d66e66c07964a5c2fe339ad1a83cb0fdf63045aa246c6ff43995464a2638c"
   }
}
```

**Comment by roman-khimov** [comment]
onNEP17Payment of 12f8d0d32b237ece13593062883fc28e2cbb799b:
```
5837     INITSLOT      11 local, 3 arg
5840     NOP
5841     SYSCALL       System.Runtime.GetCallingScriptHash (39536e3c)
5846     STLOC0
5847     PUSH1
5848     CONVERT       Boolean (20)
5850     STLOC1
5851     LDLOC0
5852     PUSHDATA1     cf76e28bd0062c4a478ee35561011319f3cfa4d2
5874     EQUAL
5875     JMPIFNOT_L    5888 (13/0d000000)
5880     PUSH1
5881     CONVERT       Boolean (20)
5883     JMP_L         5913 (30/1e000000)
5888     NOP
5889     LDLOC0
5890     PUSHDATA1     f563ea40bc283d4d0e05c48ea305b3f2a07340ef
5912     EQUAL
5913     NOP
5914     JMPIFNOT_L    5927 (13/0d000000)
5919     NOP
5920     PUSH0
5921     CONVERT       Boolean (20)
5923     DUP
5924     STLOC1
5925     DROP
5926     NOP
5927     NOP
5928     LDLOC1
5929     JMPIFNOT_L    5935 (6/06000000)
5934     ABORT
5935     NOP
5936     LDLOC0
5937     PUSHDATA1     cf76e28bd0062c4a478ee35561011319f3cfa4d2
5959     EQUAL
5960     JMPIF_L       5973 (13/0d000000)
5965     PUSH0
5966     CONVERT       Boolean (20)
5968     JMP_L         5977 (9/09000000)
5973     NOP
5974     LDARG2
5975     PUSHNULL
5976     NOTEQUAL
5977     NOP
5978     JMPIF_L       5991 (13/0d000000)
5983     PUSH0
5984     CONVERT       Boolean (20)
5986     JMP_L         5999 (13/0d000000)
5991     NOP
5992     LDARG1
5993     PUSHINT32     5000000 (404b4c00)
5998     GE
5999     NOP
6000     JMPIFNOT_L    6252 (252/fc000000)
6005     NOP
6006     SYSCALL       System.Runtime.GetInvocationCounter (84271143)
6011     PUSH1
6012     GT
6013     JMPIFNOT_L    6019 (6/06000000)
6018     ABORT
6019     NOP
...
```

**Comment by roman-khimov** [comment]
Looks like it's 7b713762be8262f63a4756b8ddf608862d141d08 from #2270.

**Comment by roman-khimov** [comment]
Doesn't reproduce on #2270.

**Comment by roman-khimov** [comment]
Fixed by #2270.
